ActivatePubKey/handler.ts

```diff
@@ -0,0 +1,202 @@
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import { Context } from "@azure/functions";
+import { RequiredParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_param";
+import { RequiredBodyPayloadMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorValidation,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+import * as express from "express";
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as O from "fp-ts/lib/Option";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { JwkPublicKeyFromToken } from "@pagopa/ts-commons/lib/jwk";
+import { ActivatedPubKey } from "../generated/definitions/internal/ActivatedPubKey";
+import { AssertionRef } from "../generated/definitions/internal/AssertionRef";
+import { ActivatePubKeyPayload } from "../generated/definitions/internal/ActivatePubKeyPayload";
+import { AssertionWriter, PopDocumentWriter } from "../utils/writers";
+import { PopDocumentReader } from "../utils/readers";
+import { JwkPubKeyHashAlgorithmEnum } from "../generated/definitions/internal/JwkPubKeyHashAlgorithm";
+import {
+  AssertionFileName,
+  PendingLolliPopPubKeys,
+  TTL_VALUE_AFTER_UPDATE
+} from "../model/lollipop_keys";
+import { PubKeyStatusEnum } from "../generated/definitions/internal/PubKeyStatus";
+import {
+  retrievedLollipopKeysToApiActivatedPubKey,
+  retrievedValidPopDocument
+} from "../utils/lollipopKeys";
+import {
+  getAlgoFromAssertionRef,
+  getAllAssertionsRef
+} from "../utils/lollipopKeys";
+import { domainErrorToResponseError } from "../utils/errors";
+
+type ActivatePubKeyHandler = (
+  context: Context,
+  assertion_ref: AssertionRef,
+  body: ActivatePubKeyPayload
+) => Promise<
+  | IResponseSuccessJson<ActivatedPubKey>
+  | IResponseErrorNotFound
+  | IResponseErrorValidation
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+>;
+
+export const ActivatePubKeyHandler = (
+  popDocumentReader: PopDocumentReader,
+  popDocumentWriter: PopDocumentWriter,
+  assertionWriter: AssertionWriter
+): ActivatePubKeyHandler => (
+  _,
+  assertion_ref,
+  body
+): ReturnType<ActivatePubKeyHandler> =>
+  pipe(
+    popDocumentReader(assertion_ref),
+    TE.mapLeft(domainErrorToResponseError),
+    TE.chainW(
+      TE.fromPredicate(PendingLolliPopPubKeys.is, () =>
+        ResponseErrorInternal("Unexpected status on pop document")
```

Comment:
add logs on internal errors

Reply:
Done in [2d5ffd0](https://github.com/pagopa/io-functions-lollipop/pull/13/commits/2d5ffd0ca6711be255f0ddb2a234805dac04c6de)